src/CalendarHTMLFormatter.php

```diff
@@ -24,6 +26,11 @@ final class CalendarHTMLFormatter implements CalendarFormatterInterface
      */
     private $mapping;
 
+    /**
+     * @var FormatterMiddleware
+     */
+    private $middleware;
```

Comment:
I'm probably overlooking something, but if this is one instance of `FormatterMiddleware`, how would we add more middlewares, for example for sold out, later?

Maybe it won't be necessary, but then the concept of the middleware is maybe a bit overengineered and it's basically an if/else check behind an abstraction? 🤔 

Alternatively a decorator pattern could make this cleaner than a literal if/else inside `CalendarHTMLFormatter` and `CalendarPlainTextFormatter`, but less complex than the middleware with a closure (to read)?

In any case this is not _that_ important to me, after all it's an implementation detail and can always be refactored later if ever needed. Just trying to challenge your solution since you were in doubt yourself 😄

Reply:
In any case it's a clever solution to not have to duplicate the logic in every formatter 👍

Reply:
Resolving this based on https://github.com/cultuurnet/calendar-summary-v3/pull/36#discussion_r564491827